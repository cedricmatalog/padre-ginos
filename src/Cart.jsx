import { intl } from "./utils";

const Cart = ({ cart }) => {
  return (
    <div className="cart">
      <h2>Cart</h2>
      {cart.length === 0 ? (
        <p>Your cart is empty</p>
      ) : (
        <ul>
          {cart.map((item, index) => (
            <li key={index}>
              {item.pizza.name} ({item.size}) - {intl.format(item.price)}
            </li>
          ))}
        </ul>
      )}
    </div>
  );
};

export default Cart;
