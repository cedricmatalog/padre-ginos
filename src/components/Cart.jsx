import React from "react";
import { intl } from "../utils";

const Cart = ({ cart, checkout }) => {
  const totalPrice = cart.reduce((total, item) => total + item.price, 0);

  return (
    <div className="cart">
      <h2>Cart</h2>
      {cart.length === 0 ? (
        <p>Your cart is empty</p>
      ) : (
        <ul>
          {cart.map((item, index) => (
            <li key={`${item.pizza.id}-${item.size}-${index}`}>
              <span className="size">{item.size}</span> -
              <span className="type">{item.pizza.name}</span> -
              <span className="price">{intl.format(item.price)}</span>
            </li>
          ))}
        </ul>
      )}
      <p>Total: {intl.format(totalPrice)}</p>
      <button onClick={checkout} disabled={cart.length === 0}>
        Checkout
      </button>
    </div>
  );
};

export default Cart;
