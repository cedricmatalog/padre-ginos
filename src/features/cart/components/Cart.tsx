import { formatPrice } from "../../../shared/utils/formatting";
import type { CartProps } from "../../../shared/types";

const Cart: React.FC<CartProps> = ({ cart, checkout }) => {
  const totalPrice = cart.reduce((total, item) => total + item.price, 0);
  const totalItems = cart.reduce((total, item) => total + (item.quantity || 1), 0);

  return (
    <aside className="bg-padre-background rounded-lg shadow-sm p-6 sticky top-8 h-fit border border-padre-border transition-colors duration-200" aria-labelledby="cart-heading">
      <header className="mb-6">
        <h2 id="cart-heading" className="font-pacifico text-2xl text-padre-primary mb-4">
          Your Cart
        </h2>
        <div className="w-full h-px bg-padre-border"></div>
      </header>
      
      {cart.length === 0 ? (
        <div className="text-center py-8" role="status">
          <p className="text-padre-muted">Your cart is empty</p>
        </div>
      ) : (
        <section aria-label="Cart items">
          <ul className="space-y-4 mb-6" role="list">
            {cart.map((item, index) => (
              <li 
                key={`${item.pizza.id}-${item.size}-${index}`} 
                className="pb-4 border-b border-padre-border last:border-b-0"
                role="listitem"
              >
                <div className="flex justify-between items-start">
                  <div className="flex-1">
                    <div className="flex items-center gap-2 mb-1">
                      <span className="bg-padre-light text-padre-muted px-2 py-1 rounded text-xs font-medium">
                        {item.size === 'S' ? 'Small' : item.size === 'M' ? 'Medium' : 'Large'}
                      </span>
                      {(item.quantity || 1) > 1 && (
                        <span className="bg-padre-secondary text-white px-2 py-1 rounded text-xs font-medium">
                          x{item.quantity || 1}
                        </span>
                      )}
                    </div>
                    <h3 className="font-medium text-padre-primary">{item.pizza.name}</h3>
                  </div>
                  <data value={item.price} className="font-medium text-padre-primary">
                    {formatPrice(item.price)}
                  </data>
                </div>
              </li>
            ))}
          </ul>
        </section>
      )}
      
      <footer>
        <div className="bg-padre-light rounded-lg p-4 mb-4">
          <div className="flex justify-between items-center">
            <span className="font-medium text-padre-primary">Total:</span>
            <data value={totalPrice} className="font-semibold text-xl text-padre-primary">
              {formatPrice(totalPrice)}
            </data>
          </div>
          <span className="sr-only">Total price: {formatPrice(totalPrice)}</span>
        </div>
        <button 
          className="btn-secondary w-full disabled:bg-padre-muted disabled:cursor-not-allowed"
          onClick={checkout} 
          disabled={cart.length === 0}
          aria-describedby="checkout-help"
        >
          Checkout {totalItems > 0 && `(${totalItems})`}
        </button>
        <div id="checkout-help" className="sr-only">
          Proceed to payment and complete your order
        </div>
      </footer>
    </aside>
  );
};

export default Cart;
