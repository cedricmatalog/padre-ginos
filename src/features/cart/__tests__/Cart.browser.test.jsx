import { expect, test } from "vitest";
import { render } from "vitest-browser-react";
import Cart from "../components/Cart";

test("snapshot with nothing in cart", () => {
  const { asFragment } = render(<Cart cart={[]} checkout={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});
