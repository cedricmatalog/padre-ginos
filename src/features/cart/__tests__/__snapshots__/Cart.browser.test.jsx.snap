// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot with nothing in cart 1`] = `
<DocumentFragment>
  <aside
    aria-labelledby="cart-heading"
    class="bg-padre-background rounded-lg shadow-sm p-6 sticky top-8 h-fit border border-padre-border transition-colors duration-200"
  >
    <header
      class="mb-6"
    >
      <h2
        class="font-pacifico text-2xl text-padre-primary mb-4"
        id="cart-heading"
      >
        Your Cart
      </h2>
      <div
        class="w-full h-px bg-padre-border"
      />
    </header>
    <div
      class="text-center py-8"
      role="status"
    >
      <p
        class="text-padre-muted"
      >
        Your cart is empty
      </p>
    </div>
    <footer>
      <div
        class="bg-padre-light rounded-lg p-4 mb-4"
      >
        <div
          class="flex justify-between items-center"
        >
          <span
            class="font-medium text-padre-primary"
          >
            Total:
          </span>
          <data
            class="font-semibold text-xl text-padre-primary"
            value="0"
          >
            $0.00
          </data>
        </div>
        <span
          class="sr-only"
        >
          Total price: $0.00
        </span>
      </div>
      <button
        aria-describedby="checkout-help"
        class="btn-secondary w-full disabled:bg-padre-muted disabled:cursor-not-allowed"
        disabled=""
      >
        Checkout 
      </button>
      <div
        class="sr-only"
        id="checkout-help"
      >
        Proceed to payment and complete your order
      </div>
    </footer>
  </aside>
</DocumentFragment>
`;
